### ISSUE TITLE
[SSL: CERTIFICATE_VERIFY_FAILED]

### ISSUE THREAD

**Issue by harseth** [rss2email/rss2email#201]
I am running rss2email on a windows server and I can see an error. I have checked the certificate and it's valid. I have also checked the root cert and it's valid to 2035. 

[ERROR] <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1076)>: SRVarmland (https://api.sr.se/api/rss/program/93.

**Comment by auouymous** [comment]
Is the time correct on the windows server?

**Comment by harseth** [comment]
Yes, the time is correct on the server.

**Comment by Ekleog** [comment]
The certificate chain includes Let's Encrypt certificates, which have been known to cause trouble recently with the expiration of the cross-signature, which if I can read correctly is currently the one in use (the website reports the ISRG root then LE's R3 cert). If I remember correctly, ISRG's signature on LE's R3 cert expired recently, thus causing the issues.

I'm sorry but there's nothing that can be done by rss2email here. This is a matter of python's certificate store on your windows server. I'm not sure how it's being handled, but it should probably contain LE's R3 cert as a root cert, as acme still defaults to renewing cert chains with this expired signature due to some old phones that don't check cert expiration and also don't have LE's new root (that should be present on newer servers).

TL;DR: please check that python running on your windows server can access the website using python's usual webpage-fetching tools. It probably won't work, so you'll have to figure out a way to configure your windows server's python installation so that it does.

**Comment by auouymous** [comment]
The following command should display a 200 status code.

```
python -c "import requests; print(requests.get('https://api.sr.se/api/rss/program/93'))"
```

**Comment by harseth** [comment]
> The following command should display a 200 status code.
> 
> ```
> python -c "import requests; print(requests.get('https://api.sr.se/api/rss/program/93'))"
> ```

When i run this command on the server i get:
<Response [200]>

**Comment by harseth** [comment]
> The certificate chain includes Let's Encrypt certificates, which have been known to cause trouble recently with the expiration of the cross-signature, which if I can read correctly is currently the one in use (the website reports the ISRG root then LE's R3 cert). If I remember correctly, ISRG's signature on LE's R3 cert expired recently, thus causing the issues.
> 
> I'm sorry but there's nothing that can be done by rss2email here. This is a matter of python's certificate store on your windows server. I'm not sure how it's being handled, but it should probably contain LE's R3 cert as a root cert, as acme still defaults to renewing cert chains with this expired signature due to some old phones that don't check cert expiration and also don't have LE's new root (that should be present on newer servers).
> 
> TL;DR: please check that python running on your windows server can access the website using python's usual webpage-fetching tools. It probably won't work, so you'll have to figure out a way to configure your windows server's python installation so that it does.

This was my theory also, so I was trying to see if the server was missing any root cert and if they did update correctly. Like you describe in the text is that one of the root certs for Let's Encrypt is expired. I did find some info about issue. the DST Root CA X3 has expired 2021-03-17. I got the new ISRG Root X1. When I checked the cert for https://api.sr.se/api/rss/program/93 I can see it is signed with ISRG Root X1 that is valid to 2035-06-04. I can't find any sign that the cert is signed with the old one.

**Comment by auouymous** [comment]
Does this also return a 200?

```
python -c "import urllib.request; print(urllib.request.urlopen('https://api.sr.se/api/rss/program/93').getcode())"
```

And does this return JSON data?

```
python -c "import feedparser; print(feedparser.parse('https://api.sr.se/api/rss/program/93'))"
```

**Comment by harseth** [comment]
` python -c "import urllib.request; print(urllib.request.urlopen('https://api.sr.se/api/rss/program/93').getcode())"` 
This give med 200.
`python -c "import feedparser; print(feedparser.parse('https://api.sr.se/api/rss/program/93'))"`
This give me json data correct.

**Comment by harseth** [comment]
I think my error is now gone. I did some more digging in my root cert. I did find that I did have ISRG Root X1 but not
ISRG Root X2. So I downloaded that one and then added the missing cert in the root cert. Then I also added Let’s Encrypt R3 in the map Intermediate Certificate. Then I deleted the expired root cert DST Root CA X3. I did a reboot and now I can´t see the SSL error. This fixes my issue. You can read more here: 
https://www.stephenwagner.com/2021/09/30/sophos-dst-root-ca-x3-expiration-problems-fix/. 

And thanks to all that responded to my issue here. Hopefully, this can help another rss2email user that has this issue.
